Financial-Times/n-messaging-client
# UG-823: Add EmbeddedOnboarding messages

**Description**

The US Growth team wants to launch an embedded version of the onboarding journey as an on-site message.
As explained in [this PR](https://github.com/Financial-Times/next-messaging-guru/pull/167), we want to show the message only to users on the `onboardingHomepage` flag's variant. The status of the flag will be served as a [component](https://github.com/Financial-Times/next-messaging-guru/pull/167/files#diff-1b942eb097fd80f71ce80ef8ab70b973e9a6063288ede6a193e77f938890b97aR6) from `next-messaging-guru`.

This PRs adds a template for the `EmbeddedOnboarding` message configuring it as lazy so that it can consume the data coming from `next-messaging-guru`. The template includes a veil to increase the opacity of the background and a container for the message markup that will be served as well through the `next-messaging-guru`'s component.

Both the veil and the container [will be display only](https://github.com/Financial-Times/n-messaging-client/pull/426/files#diff-d175f6163f6133f22dceb03e22568f12d0541ef440cfb1cb7446dcfa7e508febR55) if next-messaging-guru response sends the onboardingHomepage flag status as **true**

This PR also replaces `node-sass` with `dart-sass` to fix the issues with [index.scss resolution](https://github.com/sass/node-sass/issues/2339)  that came up on the Sass build after the origami migration to NPM

**Ticket**
https://financialtimes.atlassian.net/browse/UG-823

**How to test this code?**

- Check out this branch and run it locally with `demo-run`. If the [next-messaging-guru change](https://github.com/Financial-Times/next-messaging-guru/pull/167) hasn't been released, you'll need to check out that code locally as well, start it and then run n-messaging-client with `demo-with-guru` instead.
- Go to [FT Toggler](https://toggler.ft.com/) and activate the `EmbeddedOnboarding` variant on [messageSlotBottom](https://toggler.ft.com/#messageSlotBottom) and `variant` on [homepageOnboarding](https://toggler.ft.com/#homepageOnboarding ) 
- Go to the [local demo route](https://local.ft.com:5005/) for `n-messaging-client`, and you should see the embedded onboarding message. You should finish the journey, and the message should disappear.
- Go back to [FT Toggler](https://toggler.ft.com/) and set the [homepageOnboarding](https://toggler.ft.com/#homepageOnboarding) flag to `control`. Go back to the [local demo route](https://local.ft.com:5005/) and confirm that the message doesn't show
